plugins-bundled/internal/marketplace-app/src/components/InstallControls.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState } from 'react';
+import { css } from '@emotion/css';
+import { gt, satisfies } from 'semver';
+
+import { config } from '@grafana/runtime';
+import { Button, HorizontalGroup, Icon, Select, useStyles2 } from '@grafana/ui';
+import { AppEvents, GrafanaTheme2, OrgRole } from '@grafana/data';
+
+import { Metadata, Plugin } from '../types';
+import { hasRole } from '../helpers';
+import API from '../api';
+
+// This isn't exported in the sdk yet
+// @ts-ignore
+import appEvents from 'grafana/app/core/app_events';
+
+interface Props {
+  localPlugin?: Metadata;
+  remotePlugin: Plugin;
+  slug: string;
+}
+
+export const InstallControls = ({ localPlugin, remotePlugin, slug }: Props) => {
+  const [arch, setArch] = useState<string | undefined>();
+  const [loading, setLoading] = useState(false);
+  const [isInstalled, setIsInstalled] = useState(!!localPlugin);
+  const [shouldUpdate, setShouldUpdate] = useState(
+    remotePlugin?.version && localPlugin?.info.version && gt(remotePlugin?.version!, localPlugin?.info.version!)
+  );
+
+  const styles = useStyles2(getStyles);
+  const api = new API();
```

Comment:
I see API bein instantiated in few places. Maybe you can use a singleton service pattern we use in the core services? the `new API...` would become `getMarketplaceAPI().XXX`

Reply:
I decided to replace it with functions exported under a namespace. I think the singleton pattern would be ideal if there was some sort of internal state required by the methods (which there was when it had a constructor).